Framework/Kernel/inc/MantidKernel/Strings.h

```diff
@@ -360,11 +360,9 @@ template <typename Integer> std::vector<std::vector<Integer>> parseGroups(const
     const auto tokens = Kernel::StringTokenizer(
         str, "+", Kernel::StringTokenizer::TOK_TRIM | Kernel::StringTokenizer::TOK_IGNORE_EMPTY);
     std::vector<Integer> group;
-    group.reserve(tokens.count());
-    for (const auto &t : tokens) {
-      // add this number to the group we're about to add
-      group.emplace_back(boost::lexical_cast<Integer>(t));
-    }
+    group.reserve(tokens.count() + 1);
```

Comment:
Could you help me understand the `+1`?

Reply:
I noticed we added n `tokens` then emplaced back `group` straight after for n+1
Do you think it's worth a comment in this case? I was on the fence about it

Reply:
Is the `emplace_back` not into `groups` (plural) and not `group`. Admitedley that's hard to parse too...

Reply:
Good point, I'll rename the variables as it's subtle :+1: